File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):

    stack = []
    for i in s:
        if i == "(":
            stack.append(")")
        elif i == ")":
            if len(stack) == 0:
                return False
            elif stack[0] == ")":
                stack.pop()
        else:
            return False
    if len(stack) != 0:
        return False
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int prev = 0;
        for(char c : s.toCharArray()){
            if(c == '('){
                prev ++;
            }else {
                prev --;
            }
            if(prev < 0){
                return false;
            }
        }

        return prev == 0;
    }
}